ServerCore/PuzzleStateHelper.cs

```diff
@@ -294,6 +294,38 @@ public static async Task SetUnlockStateAsync(PuzzleServerContext context, Event
             await context.SaveChangesAsync();
         }
 
+        private static Dictionary<int, DateTime> TimedUnlockExpiryCache = new Dictionary<int, DateTime>();
```

Comment:
Good point. But since my newer design has a global throttle that is outside the dictionary, I'm going with a lock so I can keep it all consistent. The expensive stuff happens outside the lock, so I think it'll be OK.